src/Symfony/Component/Message/MessageBusInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+interface MessageBusInterface
+{
+    /**
+     * Handle the message.
+     *
+     * The bus can return a value coming from handlers, but is not required to do so.
+     *
+     * @param object $message
+     *
+     * @return mixed
```

Comment:
And that's the big point here: that's not "just" a command bus but it is a message bus. This can also be used for queries in a CQRS-type architecture. 

Note that if you want to enforce such limitation for commands, you can very easily create your own instance of the `MessageBus` and have a special "noop returns" (or  "EnforceCommandsDoNotReturnAnything") middleware that would force return `null` for example.

Reply:
Ok, thank for explanations.